### DISCUSSION TITLE
Dreame D10s (no Pro) support

### DISCUSSION THREAD

**Discussion by dborrajo** [merdok/homebridge-miot#542] [Q&A]
Is there any possibility of adding support for this robot cleaner

**Comment by merdok** [comment]
Hi, did you already try to add the device to the plugin? Can you clarify what is not working?

**Reply by dborrajo** [reply]
How could I do it? The only way to do that is to get the device token and IP, isn't it? And Dreamehome App does not provide this information.

**Reply by merdok** [reply]
Yeah I mean you have to get the token for your device. I you cannot add the device to the xiaomi home app then I am not sure what other possibilities you have. Maybe there are some ways by now to get the token from the dreame home app?

**Reply by dborrajo** [reply]
Dreame App does not provide this information and there is no option to MiHome App to recognize the device. All new models of Dreametech sync with this App.

I hope something could be done soon.

Thks

**Reply by merdok** [reply]
Until somebody finds out how to get the token from those devices, there is unfortunately nothing that can be done from my side...

**Reply by dborrajo** [reply]
Hey, I think it could be useful: https://github.com/Tasshack/dreame-vacuum

**Reply by merdok** [reply]
Thanks for that, did a quick look and it seems the connection is to the xiaomi servers the same as with this plugin so not quite certain why you cannot find your device in this he cloud.

**Reply by dborrajo** [reply]
But it seems to have other endpoints (https://github.com/Tasshack/dreame-vacuum/discussions/109). Is there anything I could do to help you?

**Reply by merdok** [reply]
Yeah i mean they mention them there, but i cannot find any place in the code where they are used. Or am I missing something?

**Reply by dborrajo** [reply]
I have been doing some research decompiling the Dreametech App and I think the login page is https://smarthomecn.dreame.tech/. This one uses some kind of oAuth which I am not able to figure it out.

**Reply by merdok** [reply]
You could try to replace the url in MiCloud protocol with this one and see if maybe that would be enough...

**Reply by Jeremesk** [reply]
Hi, has someone managed to do something with the Dreamtech app URL, or from Tasshack's implementation in Home Assistant?

**Reply by merdok** [reply]
I personally did not have a look at it yet...
